src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java

```diff
@@ -0,0 +1,262 @@
+import static org.opensearch.security.Song.TITLE_MAGNUM_OPUS;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.BASIC_AUTH_DOMAIN_ORDER;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+import static org.opensearch.test.framework.cluster.SearchRequestFactory.queryStringQueryRequest;
+import static org.opensearch.test.framework.matcher.ExceptionMatcherAssert.assertThatThrownBy;
+import static org.opensearch.test.framework.matcher.OpenSearchExceptionMatchers.statusException;
+import static org.opensearch.test.framework.matcher.SearchResponseMatchers.isSuccessfulSearchResponse;
+import static org.opensearch.test.framework.matcher.SearchResponseMatchers.numberOfTotalHitsIsEqualTo;
+import static org.opensearch.test.framework.matcher.SearchResponseMatchers.searchHitContainsFieldWithValue;
+import static org.opensearch.test.framework.matcher.SearchResponseMatchers.searchHitsContainDocumentWithId;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class JwtAuthenticationTests {
+
+	public static final String CLAIM_USERNAME = "preferred-username";
+	public static final String CLAIM_ROLES = "backend-user-roles";
+
+	public static final String USER_SUPERHERO = "superhero";
+	public static final String USERNAME_ROOT = "root";
+	public static final String ROLE_ADMIN = "role_admin";
+	public static final String ROLE_DEVELOPER = "role_developer";
+	public static final String ROLE_QA = "role_qa";
+	public static final String ROLE_CTO = "role_cto";
+	public static final String ROLE_CEO = "role_ceo";
+	public static final String ROLE_VP = "role_vp";
+	public static final String POINTER_BACKEND_ROLES = "/backend_roles";
+	public static final String POINTER_USERNAME = "/user_name";
+
+	public static final String QA_DEPARTMENT = "qa-department";
+
+	public static final String CLAIM_DEPARTMENT = "department";
+
+	public static final String DEPARTMENT_SONG_INDEX_PATTERN = String.format("song_lyrics_${attr.jwt.%s}", CLAIM_DEPARTMENT);
+
+	public static final String QA_SONG_INDEX_NAME = String.format("song_lyrics_%s", QA_DEPARTMENT);
+
+	private static final KeyPair KEY_PAIR = Keys.keyPairFor(SignatureAlgorithm.RS256);
+	private static final String PUBLIC_KEY = new String(Base64.getEncoder().encode(KEY_PAIR.getPublic().getEncoded()));
+
+	static final TestSecurityConfig.User ADMIN_USER = new TestSecurityConfig.User("admin").roles(ALL_ACCESS);
+
+	private static final String JWT_AUTH_HEADER = "jwt-auth";
+
+	private static final JwtAuthorizationHeaderFactory tokenFactory = new JwtAuthorizationHeaderFactory(
+		KEY_PAIR.getPrivate(),
+		CLAIM_USERNAME,
+		CLAIM_ROLES,
+		JWT_AUTH_HEADER);
+
+	public static final TestSecurityConfig.AuthcDomain JWT_AUTH_DOMAIN = new TestSecurityConfig
+		.AuthcDomain("jwt", BASIC_AUTH_DOMAIN_ORDER - 1)
```

Comment:
why `-1` here and not leave it to `0`?

Reply:
The point here is to insert `JWT_AUTH_DOMAIN` before the basic auth domain. It is also possible to use order 0 for JWT auth and 1 for basic auth.

Reply:
okay, got it. thank you!

Reply:
anytime :)